src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,347 @@
+    type = int)
+    parser.add_argument("--output", help = "Output directory path", default = settings.NUMBER_DIR)
+    parser.add_argument("--operand-count", help = "Number of maximum operands in an expression",
+    nargs = 1, default = [OPERAND_COUNT_MAX], choices = range(OPERAND_COUNT_MIN, OPERAND_COUNT_MAX +  1), type = int)
+    parser.add_argument("--seed", help = "Seed to used while generating random number", nargs = 1, default = [DEFAULT_SEED],
+    type = int)
+    if len(sys.argv) == 1:
+        parser.print_help()
+        sys.exit(1)
+
+    script_args = parser.parse_args()
+
+    return script_args
+
+class GenSequence:
+    def __init__(self, options, bit):
+        bitcount = 0
+        arithcount = 0
+        num_operands = options.operand_count[0]
+        self.num_operands = num_operands
+        self.operators_list = []
+        for op in options.op:
+            if (not bit):
+                if (op == "+"):
+                    self.operators_list.append(self.add)
+                    arithcount += 1
+                elif (op == "-"):
+                    self.operators_list.append(self.sub)
+                    arithcount += 1
+                elif (op == "*"):
+                    self.operators_list.append(self.mul)
+                    arithcount += 1
+                elif (op == "div"):
+                    self.operators_list.append(self.div)
+                    arithcount += 1
+                elif (op == "%"):
+                    self.operators_list.append(self.mod)
+                    arithcount += 1
+            else:
+                if (op == "^"):
+                    self.operators_list.append(self.bitxor)
+                    bitcount += 1
+                elif (op == "&"):
+                    self.operators_list.append(self.bitand)
+                    bitcount += 1
+                elif (op == "|"):
+                    self.operators_list.append(self.bitor)
+                    bitcount += 1
+                elif (op == "<<"):
+                    self.operators_list.append(self.bitshiftleft)
+                    bitcount += 1
+                elif (op == ">>"):
+                    self.operators_list.append(self.bitshiftright)
+                    bitcount += 1
+        self.operands = [0] * num_operands # Creating a list for operands according to number of operands
+        self.operators = [0] * (num_operands - 1) # Creating a list for operators according to number of operands
+
+        MAXNUM = MAX
+        MINNUM = MIN
+        if (bit): # Setting minimum and maximum according to bitwise or arithmetic calculations
+            MAXNUM = BITMAX
+            MINNUM = BITMIN
+        else:
+            MAXNUM = MAX
+            MINNUM = MIN
+
+        # Generating the expression
+        for i in range(num_operands - 1):
+            self.operands[i] = random.randint(MINNUM, MAXNUM)
+            if (arithcount != 1):
```

Comment:
There is no need for `arithcount` use `not bit and len(options.op)` instead. In addition I guess the `len(options.op)` can be avoid as well.